src/render/painter.js

```diff
@@ -344,48 +347,77 @@ class Painter {
             return;
         }
 
+        const renderableSkippedTileIDs = [];
         if (this._tileClippingMaskIDs && !this.terrain) {
             let dirtyStencilClippingMasks = false;
             // Equivalent tile set is already rendered in stencil
             for (const coord of tileIDs) {
                 if (this._tileClippingMaskIDs[coord.key] === undefined) {
                     dirtyStencilClippingMasks = true;
-                    break;
+                }
+                if (this._skippedStencilTileIDs.includes(coord.key)) {
+                    const tile = sourceCache.getTile(coord);
+                    const bucket = (tile.getBucket(layer): any);
+                    if (!bucket) {
+                        continue;
+                    }
+                    this._skippedStencilTileIDs = this._skippedStencilTileIDs.filter(item => item !== coord.key);
```

Comment:
Nit: `includes` + `filter` will iterate over the array twice and have a redundant array allocation. Instead you could do `indexOf` first, and then use that index to remove the coord with `splice`.